rotkehlchen/chain/ethereum/modules/blur/decoder.py

```diff
@@ -0,0 +1,118 @@
+
+BLUR_DEPOSITED: Final = b'\xe1\xff\xfc\xc4\x92=\x04\xb5Y\xf4\xd2\x9a\x8b\xfcl\xda\x04\xeb[\r<F\x07Q\xc2@,\\\\\xc9\x10\x9c'  # noqa: E501
+BLUR_AIRDROP_2_CLAIM: Final = b'\xd8\x13\x8f\x8a?7|RY\xcaT\x8ep\xe4\xc2\xde\x94\xf1)\xf5\xa1\x106\xa1[iQ<\xba+Bj'  # noqa: E501
+
+
+class BlurDecoder(DecoderInterface):
+    def __init__(
+            self,
+            evm_inquirer: 'EthereumInquirer',
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator',
+    ) -> None:
+        super().__init__(
+            evm_inquirer=evm_inquirer,
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+        )
+
+    def _decode_events(
+            self,
+            transaction: 'EvmTransaction',
+            decoded_events: list['EvmEvent'],
+            all_logs: list['EvmTxReceiptLog'],
+    ) -> list['EvmEvent']:
+        """This function is used to decode the staking events done via Blur."""
+        for event in decoded_events:
+            if (
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.asset == A_BLUR
+            ):
+                event.event_type = HistoryEventType.STAKING
+                event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                event.counterparty = CPT_BLUR
+                event.notes = f'Stake {event.balance.amount} BLUR to {BLUR_STAKING_CONTRACT}'
+                break
+        else:
+            for tx_log in all_logs:
+                if tx_log.topics[0] == BLUR_DEPOSITED:
+                    self._decode_stake(DecoderContext(
+                        transaction=transaction,
+                        all_logs=all_logs,
+                        decoded_events=decoded_events,
+                        action_items=[],
+                        tx_log=tx_log,
+                    ))
+                    break
+
+        return decoded_events
+
+    def _decode_stake(self, context: DecoderContext) -> DecodingOutput:
+        """Decode staking event in the Blur protocol"""
+        if self.base.is_tracked(user_address := hex_or_bytes_to_address(context.tx_log.topics[1])) is False:  # noqa: E501
+            return DEFAULT_DECODING_OUTPUT
+
+        stake_amount = hex_or_bytes_to_int(context.tx_log.data[:32])
+        stake_amount_norm = token_normalized_value_decimals(
+            token_amount=stake_amount,
+            token_decimals=DEFAULT_TOKEN_DECIMALS,
+        )
+        context.decoded_events.append(self.base.make_event_from_transaction(
+            transaction=context.transaction,
+            tx_log=context.tx_log,
+            event_type=HistoryEventType.STAKING,
+            event_subtype=HistoryEventSubType.DEPOSIT_ASSET,
+            asset=A_BLUR,
+            balance=Balance(amount=stake_amount_norm),
+            location_label=user_address,
+            notes=f'Stake {stake_amount_norm} BLUR to {BLUR_STAKING_CONTRACT}',
```

Comment:
same comment about the contract. I believe it is information not relevant to the user